### PR TITLE
Refactor pkg/subctl/cmd/cloud -> pkg/cloud just for AWS

### PR DESCRIPTION
**Pull request by Jaanki**
Signed-off-by: Janki Chhatbar <jchhatba@redhat.com>

### CODE REVIEW

**Review by vthapar — DISMISSED** [review]

File: pkg/subctl/cmd/cloud/prepare/aws.go [hunk 1]
```diff
@@ -20,67 +20,49 @@ package prepare
...
 	"github.com/submariner-io/submariner-operator/internal/cli"
 	"github.com/submariner-io/submariner-operator/internal/exit"
+	"github.com/submariner-io/submariner-operator/internal/restconfig"
+	"github.com/submariner-io/submariner-operator/pkg/cloud"
+	cloudaws "github.com/submariner-io/submariner-operator/pkg/cloud/aws"
+	"github.com/submariner-io/submariner-operator/pkg/cloud/prepare"
 	"github.com/submariner-io/submariner-operator/pkg/subctl/cmd/cloud/aws"
 )
 
+var config *cloudaws.Config
```
Comment (skitt) [inline pkg/subctl/cmd/cloud/prepare/aws.go]: This needs to declare the instance itself, not just a pointer to it:

```suggestion
var config cloudaws.Config
```

Then whenever you need to pass `config` around, pass a pointer to it, `&config`.

**Review by skitt — CHANGES_REQUESTED** [review]

File: pkg/cloud/aws/aws.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	Region          string
+	Profile         string
+	CredentialsFile string
+	OcpMetadataFile string
+	GWInstanceType  string
+}
+
+// RunOnAWS runs the given function on AWS, supplying it with a cloud instance connected to AWS and a reporter that writes to CLI.
+// The functions makes sure that infraID and region are specified, and extracts the credentials from a secret in order to connect to AWS.
+func RunOnAWS(restConfigProducer restconfig.Producer, config Config, status reporter.Interface,
```
Comment (tpantelis) [inline pkg/cloud/aws/aws.go]:
```suggestion
func RunOn(restConfigProducer restconfig.Producer, config Config, status reporter.Interface,
```

File: pkg/cloud/cleanup/aws.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+package cleanup
+
+import (
+	"github.com/submariner-io/admiral/pkg/reporter"
+	"github.com/submariner-io/cloud-prepare/pkg/api"
+	"github.com/submariner-io/submariner-operator/internal/restconfig"
+	"github.com/submariner-io/submariner-operator/pkg/cloud/aws"
+)
+
+func Aws(restConfigProducer *restconfig.Producer, status reporter.Interface) error {
```
Comment (tpantelis) [inline pkg/cloud/cleanup/aws.go]:
```suggestion
func AWS(restConfigProducer *restconfig.Producer, status reporter.Interface) error {
```

File: pkg/cloud/prepare/aws.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+import (
+	"github.com/submariner-io/admiral/pkg/reporter"
+	"github.com/submariner-io/cloud-prepare/pkg/api"
+	"github.com/submariner-io/submariner-operator/internal/restconfig"
+	"github.com/submariner-io/submariner-operator/pkg/cloud"
+	"github.com/submariner-io/submariner-operator/pkg/cloud/aws"
+)
+
+func Aws(restConfigProducer *restconfig.Producer, ports cloud.Ports, config aws.Config, status reporter.Interface) error {
```
Comment (tpantelis) [inline pkg/cloud/prepare/aws.go]:
```suggestion
func AWS(restConfigProducer *restconfig.Producer, ports cloud.Ports, config aws.Config, status reporter.Interface) error {
```

**Review by tpantelis — CHANGES_REQUESTED** [review]

**Review by tpantelis — APPROVED** [review]

**Review by skitt — APPROVED** [review]

**Pull request merged**